src/toil/leader.py

```diff
@@ -488,28 +489,29 @@ def checkForDeadlocks(self):
         """
         Checks if the system is deadlocked running service jobs.
         """
-        # If there are no updated jobs and at least some jobs issued
-        if len(self.toilState.updatedJobs) == 0 and self.getNumberOfJobsIssued() > 0:
-
-            # If all scheduled jobs are services
-            assert self.serviceJobsIssued + self.preemptableServiceJobsIssued <= self.getNumberOfJobsIssued()
-            if self.serviceJobsIssued + self.preemptableServiceJobsIssued == self.getNumberOfJobsIssued():
-
-                # Sanity check that all issued jobs are actually services
-                for jobNode in self.jobBatchSystemIDToIssuedJob.values():
-                    assert jobNode.jobStoreID in self.toilState.serviceJobStoreIDToPredecessorJob
-
-                # An active service job is one that is not in the process of terminating
-                activeServiceJobs = filter(lambda x : self.serviceManager.isActive(x), self.jobBatchSystemIDToIssuedJob.values())
```

Comment:
Yep, that is where the services that were still starting counted toward a potential deadlock. So often any service that took longer than `deadlockWait` seconds to start up would cause an exception to be thrown.

There were basically three separate problems, 1) not resetting "potential deadlock" information, 2) counting issued and non-terminated services and 3) missing deadlocks because a single issued non-service job would stop a potential deadlock, even if that job couldn't possibly run because all the cores were occupied by services. So this is where 2) arose.